cmake/modules/OmrCrossCompile.cmake

```diff
@@ -24,10 +24,14 @@ if(OMR_CROSS_COMPILE_)
 endif()
 set(OMR_CROSS_COMPILE_ 1)
 
-# Building on cygwin but using MSVC will look like cross compiling, but we don't consider it to be.
-# Note: At the moment this test is overly broad while reliable detection method is investigated.
-if(CMAKE_CROSSCOMPILING AND NOT OMR_OS_WINDOWS)
-	set(OMR_CROSSCOMPILING ON CACHE INTERNAL "")
+
+if(CMAKE_CROSSCOMPILING)
+	# Building on cygwin but using MSVC will look like cross compiling, but we don't consider it to be.
+	if(OMR_OS_WINDOWS AND "${CMAKE_HOST_SYSTEM_NAME}" MATCHES "CYGWIN|MSYS|MINGW")
```

Comment:
Suggest listing the system names in alphabetical order for easier reading (`CYGWIN|MINGW|MSYS`).

The comment should match the test, i.e. it should also mention mingw and msys.
Where is the test that we're actually using MSVC (implied by that comment)?